pkg/util/watch_filters.go

```diff
@@ -79,6 +81,38 @@ func ResolveCAPIMachineSetFromInfraMachineTemplate(namespace string) func(contex
 	}
 }
 
+// ResolveCAPIMachineFromInfraMachine resolves a CAPI Machine from an InfraMachine. It takes client.Object,
+// and uses owner references to determine the owning CAPI machine. If one is found, it returns a reconcile.Request
+// for the corresponding MAPI Machine in the MAPI namespace to trigger reconciliation of the mirror MAPI Machine.
+func ResolveCAPIMachineFromInfraMachine(namespace string) func(context.Context, client.Object) []reconcile.Request {
+	return func(ctx context.Context, obj client.Object) []reconcile.Request {
+		logger := logf.FromContext(ctx).WithValues(
+			"objectType", fmt.Sprintf("%T", obj),
+			"namespace", obj.GetNamespace(),
+			"name", obj.GetName(),
+		)
+		logger.V(4).Info("reconcile triggered")
+
+		requests := []reconcile.Request{}
+
+		for _, ref := range obj.GetOwnerReferences() {
+			gv, err := schema.ParseGroupVersion(ref.APIVersion)
+			if err != nil {
+				logger.Error(err, "Failed to parse GroupVersion", "APIVersion", ref.APIVersion)
+				continue
+			}
+
+			if ref.Kind == "Machine" && gv.Group == clusterv1.GroupVersion.Group {
+				requests = append(requests, reconcile.Request{
+					NamespacedName: client.ObjectKey{Namespace: namespace, Name: ref.Name},
+				})
+			}
```

Comment:
Do we expect multiple Machine owner references? You should probably just return at this point.

This also means you don't need to append, you can return a slice you instantiate here, initialised with a single element.